compiler/accelerators/snax.py

```diff
@@ -59,6 +61,10 @@ def lower_acc_setup(
         field_to_csr = dict(acc_op.field_items())
         ops: Sequence[Operation] = []
         for field, val in setup_op.iter_params():
+            if isinstance(val.type, builtin.IndexType):
+                val_to_i32 = arith.IndexCastOp(val, builtin.i32)
+                ops.append(val_to_i32)
+                val = val_to_i32.result
```

Comment:
I like this, but I'd prefer we split it in a more general (also parameterized) pass maybe? Can do this in a later PR.

Reply:
ok will do!